### DISCUSSION TITLE
Firebase storage file upload not working cause by cors policy.

### DISCUSSION THREAD

**Discussion by FlyingTurkman** [vercel/next.js#59865] [Help]
### Summary

I tried many methods for upload file to firebase storage and getting cors error. With Next.js 12 I developed a lot of softwares and used firebase storage. 
Here cors config file at firebase
```json
[
     {"maxAgeSeconds": 3600, "method": ["*"], "origin": ["*"]}
]
```

Is there anybody face with this issue?

### Additional information

Here error message
Access to XMLHttpRequest at 'https://firebasestorage.googleapis.com/v0/b/%22xxx.appspot.com%22/o?name=test123.jpg' from origin 'https://xxx.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

_No response_

**Comment by Braedencraig** [comment]
How are you uploading/writing your data to firebase? Might be related to that, happy to help!

**Reply by FlyingTurkman** [reply]
I figure it out problem. Its cause by environment variables at vercel. If I write variable like this
apiKey: "xxxxxx"
....

its cause of cors error

I change variables like this

apiKey: xxxxx

**Reply by Kashan-2912** [reply]
im on vite app, am having same issue. can you please tell me how to fix cor issue
Im uploading files on firestore using setDoc function

**Reply by FlyingTurkman** [reply]
My issue is about .env file. If you type apiKey like "xxxxx" this causing error you should change it like xxxx

**Reply by Kashan-2912** [reply]
WHen i saw your msg, i made env file and put all apiKeys and storageBucket etc to env file. For your info, Im using firebase. But cors issue still there when i try to upload files to firebase firestore. Any solution ?

**Reply by arewageek** [reply]
That doesn't seem to be solving mine
Changed it a couple of times and my requests still fail

**Reply by shadabali-git** [reply]
where to put this 
[
     {"maxAgeSeconds": 3600, "method": ["*"], "origin": ["*"]}
] 
is i have to create a cors.json file ??
